### PR TITLE
chore: added more examples illustrating the use of Authorization Servers (AS) and Resource Servers (RS) in this specification

### PR DESCRIPTION
**Pull request by peppelinux**
this PR is merely editorial and aims to resolve the issue https://github.com/openid/OpenID4VCI/issues/311

### CODE REVIEW

File: openid-4-verifiable-credential-issuance-1_0.md [hunk 1]
```diff
@@ -106,7 +106,7 @@ Biometrics-based Holder Binding:
...
 
 Wallet:
-:  An entity used by the Holder to request, receive, store, present, and manage Verifiable Credentials and cryptographic key material. There is no single deployment model of a Wallet: Credentials and keys can be stored and managed either locally, through a remote self-hosted service, or via a remote third-party service. In the context of this specification, the Wallet acts as an OAuth 2.0 Client (see [@!RFC6749]).
+:  An entity used by the Holder to request, receive, store, present, and manage Verifiable Credentials and cryptographic key material. There is no single deployment model of a Wallet: Credentials and keys can be stored and managed either locally, through a remote self-hosted service, or via a remote third-party service. In the context of this specification, the Wallet acts as an OAuth 2.0 Client (see [@!RFC6749]) since it obtains an Access Token to access an OAuth 2.0 Resource Server (Credential Endpoint).
```
Comment (jogu) [inline openid-4-verifiable-credential-issuance-1_0.md:109]:
```suggestion
:  An entity used by the Holder to request, receive, store, present, and manage Verifiable Credentials and cryptographic key material. There is no single deployment model of a Wallet: Credentials and keys can be stored and managed either locally, through a remote self-hosted service, or via a remote third-party service. In the context of this specification, the Wallet acts as an OAuth 2.0 Client (see [@!RFC6749]) and obtains an Access Token to access an OAuth 2.0 Resource Server (Credential Endpoint).
```

**Review by jogu — APPROVED** [review]

**Review by bc-pi — APPROVED** [review]

**Review by paulbastian — APPROVED** [review]

**Review by c2bo — APPROVED** [review]

File: openid-4-verifiable-credential-issuance-1_0.md [hunk 2]
```diff
@@ -128,7 +128,7 @@ Both the Credential and the Batch Credential Endpoints have the (optional) abili
 
 ## OAuth 2.0
 
-Every Credential Issuer utilizes an OAuth 2.0 [@!RFC6749] Authorization Server to authorize access. The same OAuth 2.0 Authorization Server can protect one or more Credential Issuers. Wallets determine the Credential Issuer's Authorization Server using the Credential Issuer's metadata (see (#credential-issuer-metadata)).
+According to the OAuth 2.0 paradigm, each Credential Issuer acts as a Resource Server that is protected by an Access Token issued by an Authorization Server, as defined in OAuth 2.0 [@!RFC6749]. The same Authorization Server may protect one or more Credential Issuers. Wallets identify the Authorization Server for a Credential Issuer by referring to the Credential Issuer's metadata (see (#credential-issuer-metadata)).
```
Comment (Sakurann) [inline openid-4-verifiable-credential-issuance-1_0.md:131]:
```suggestion
According to the OAuth 2.0 framework, each Credential Issuer acts as a Resource Server that is protected by an Access Token issued by an Authorization Server, as defined in OAuth 2.0 [@!RFC6749]. The same Authorization Server can protect one or more Credential Issuers. Wallets identify the Authorization Server for a Credential Issuer by referring to the Credential Issuer's metadata (see (#credential-issuer-metadata)).
```
let's not use lower-case may, and i think `framework` fits better than `paradigm`

File: openid-4-verifiable-credential-issuance-1_0.md [hunk 3]
```diff
@@ -206,36 +206,47 @@ Please note that the diagram does not illustrate all the optional features defin
 
 !---
 ~~~ ascii-art
```
[the remark this answers is not in the document]
Comment (peppelinux) [reply openid-4-verifiable-credential-issuance-1_0.md:208]: in this figure the CI is the first entity the wallet interacts with. In this example the discovery of the AS starts from the CI metadata

**Comment by jogu** [comment]
Discussed on today's WG call:

1. There was a consensus that splitting the credential issuer & the AS in the diagrams is a good idea
2. However we felt that the credential issuer should be to the right of the authorization server, as in most oauth sequence diagrams the resource server is shown on the right hand side of the authorization server and it was felt it would help people's mental models if we followed that convention.
3. If we're going to add AS to this diagram, then we should add it to all diagrams (I think there's only one other diagram)

**Review by jogu — CHANGES_REQUESTED** [review]
as per my above comment

**Comment by peppelinux** [comment]
I can do that during the next week, thank you for the feedback

File: openid-4-verifiable-credential-issuance-1_0.md [hunk 4]
```diff
@@ -206,36 +206,47 @@ Please note that the diagram does not illustrate all the optional features defin
...
+    |                |<------------------------------------------------------|
+    |                |                              |                        |
+    |                |  (6) Credential Request      |                        |
+    |                |      (Access Token, proof(s))|                        |
+    |                |----------------------------->|                        |
+    |                |                              |                        |
+    |                |      Credential Response     |                        |
+    |                |      with Credential(s) OR   |                        |
+    |                |      Transaction ID          |                        |
+    |                |<-----------------------------|                        |
```
Comment (peppelinux) [inline openid-4-verifiable-credential-issuance-1_0.md:249]:
```suggestion
+----------+   +-----------+            +----------------------+     +-------------------+
| End-User |   |   Wallet  |            | Authorization Server |     | Credential Issuer |
+----------+   +-----------+            +----------------------+     +-------------------+
    |                |                              |                        |
    | (1a) End-User  |                              |                        |
    |  selects       |  (1b) Credential Offer       |                        |
    |  Credential--->|  (credential type)           |                        |
    |                |<------------------------------------------------------|
    |                |                              |                        |
    |                |  (2) Obtains Issuer's        |                        |
    |                |      Credential Issuer       |                        |
    |                |      metadata                |                        |
    |                |------------------------------------------------------>|
    |                |                              |                        |
    |                |                              |  (3) Authorization     |
    |                |                              |      Request           |
    |                |                              |      (type(s) of       |
    |                |                              |      Credentials to    |
    |                |                              |      be issued)        |
    |                |----------------------------->|                        |
    |                |                              |                        |
    |  End-User Authentication / Consent            |                        |
    |                |                              |  (4) Authorization     |
    |                |                              |      Response (code)   |
    |                |<-----------------------------|                        |
    |                |                              |                        |
    |                |                              |  (5) Token Request     |
    |                |                              |      (code)            |
    |                |----------------------------->|      Token Response    |
    |                |                              |      (Access Token)    |
    |                |<-----------------------------|                        |
    |                |                              |                        |
    |                |  (6) Credential Request      |                        |
    |                |      (Access Token, proof(s))|                        |
    |                |------------------------------------------------------>|
    |                |                              |                        |
    |                |      Credential Response     |                        |
    |                |      with Credential(s) OR   |                        |
    |                |      Transaction ID          |                        |
    |                |<-----------------------------------------------------|
```

File: openid-4-verifiable-credential-issuance-1_0.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (peppelinux) [reply openid-4-verifiable-credential-issuance-1_0.md:208]: done.

**Comment by peppelinux** [comment]
@jogu I did the requested changes here: https://github.com/openid/OpenID4VCI/pull/322/commits/0521a835e0299be7d74937e33c8ecbaa6de2dcf4

**Review by Sakurann — APPROVED** [review]
thank you, Giuseppe!

**Review by jogu — APPROVED** [review]

**Pull request merged**